File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff = {}
        res = []
        for i in range(len(nums)):
            cur_diff = target - nums[i]
            if cur_diff in diff:
                res.append(diff[cur_diff])
                res.append(i)
                return res
            else:
                diff[nums[i]] = i

File: Data Structures & Algorithms/search-insert-position/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    searchInsert(nums, target) {
        // we will do a binary search to
        // get the index to insert into
        let l = 0;
        let r = nums.length - 1;
        let res = nums.length
        while(l <= r){
            let mid = Math.floor((l + r) / 2)
            if(target === nums[mid]){
                return mid;
            }else if(nums[mid] > target){
                res = mid;
                r = mid - 1;
            }else {
                l = mid + 1;
            }
        }
        return res;
    }
}
